src/lit-html.ts

```diff
@@ -220,10 +221,18 @@ export class Template {
             parent.insertBefore(document.createTextNode(strings[i]), node);
             this.parts.push(new TemplatePart('node', index++));
           }
-        } else if (!node.nodeValue!.trim()) {
-          nodesToRemove.push(node);
-          currentNode = previousNode;
-          index--;
+        } else {
+          // Strip whitespace-only nodes, only between elements, or at the
+          // beginning or end of elements.
```

Comment:
I wonder if these rules are even satisfactory to match what the browser might do with whitespace.

Reply:
Any whitespace stripping like this is going to slightly change layout. @kevinpschaaf added the original stripping as a perf optimization, and this is backing some of it out for the bug fixes. Soon I think we want to look at how we might offer some control over this, but I think this is a better middle ground.